velox/exec/Window.cpp

```diff
@@ -137,12 +137,61 @@ void Window::createWindowFunctions(
         operatorCtx_->pool(),
         &stringAllocator_));
 
-    windowFrames_.push_back(
+    auto updateKBasedFrameBounds = [&](const core::TypedExprPtr& frame,
+                                       WindowFrame& windowFrame,
+                                       bool isStartFrame) -> FrameArg {
+      if (auto constant = constantArg(frame)) {
+        auto constantOffset =
+            constant->template as<ConstantVector<int64_t>>()->valueAt(0);
+        VELOX_USER_CHECK_GE(
+            constantOffset, 1, "k in frame bounds must be at least 1");
+        return FrameArg({constantOffset, kConstantChannel, nullptr});
+      } else {
+        FlatVectorPtr<int64_t> offsets =
+            std::dynamic_pointer_cast<FlatVector<int64_t>>(
+                BaseVector::create(BIGINT(), 0, operatorCtx_->pool()));
+        return FrameArg(
```

Comment:
You don't have to explicitly invoke the constructor. Just returning the tuple should suffice.

Reply:
Thanks for the suggestion, simplified accordingly!